nexus-watcher/src/events/processor.rs

```diff
@@ -14,6 +14,47 @@ use std::path::PathBuf;
 use tokio::sync::watch::Receiver;
 use tracing::{debug, error, info};
 
+/// This implements the creation logic for [`EventProcessor`] objects
+pub struct EventProcessorFactory {
+    pub limit: u32,
+    pub files_path: PathBuf,
+    pub tracer_name: String,
+    pub moderation: Moderation,
+}
+
+impl EventProcessorFactory {
+    /// Creates a new factory instance from the provided configuration
+    pub fn from_config(config: &WatcherConfig) -> Self {
+        Self {
+            limit: config.events_limit,
+            files_path: config.stack.files_path.clone(),
+            tracer_name: config.name.clone(),
+            moderation: Moderation {
+                id: config.moderation_id.clone(),
+                tags: config.moderated_tags.clone(),
+            },
+        }
+    }
+
+    /// Builds and returns a configured [`EventProcessor`] instance.
+    /// # Arguments
+    /// - `homeserver_id`: The ID of the homeserver to process
+    pub async fn build(&self, homeserver_id: &str) -> Result<EventProcessor, DynError> {
+        let homeserver_id = PubkyId::try_from(homeserver_id).map_err(DynError::from)?;
+        let homeserver = Homeserver::get_by_id(homeserver_id)
+            .await?
+            .ok_or("Homeserver not found")?;
+
+        Ok(EventProcessor {
+            homeserver,
+            limit: self.limit,
+            files_path: self.files_path.clone(),
+            tracer_name: self.tracer_name.clone(),
+            moderation: self.moderation.clone(),
```

Comment:
To avoid cloning (each time, for each processor, for each run), maybe `EventProcessor` can have `Arc` pointers, like so:

```rust
pub struct EventProcessor {
    pub homeserver: Homeserver,
    limit: u32,
    pub files_path: Arc<PathBuf>,
    pub tracer_name: String,
    pub moderation: Arc<Moderation>,
}
```

Then its just a matter of cloning the pointers.